stable_baselines3/common/vec_env/vec_transpose.py

```diff
@@ -14,9 +17,20 @@ class VecTransposeImage(VecEnvWrapper):
     """
 
     def __init__(self, venv: VecEnv):
-        assert is_image_space(venv.observation_space), "The observation space must be an image"
+        assert is_image_space(venv.observation_space) or isinstance(
+            venv.observation_space, spaces.dict.Dict
+        ), "The observation space must be an image or dictionary observation space"
 
-        observation_space = self.transpose_space(venv.observation_space)
+        if isinstance(venv.observation_space, spaces.dict.Dict):
+            self.image_space_keys = []
+            observation_space = deepcopy(venv.observation_space)
+            for key, space in observation_space.spaces.items():
```

Comment:
here we are assuming that all image spaces follow the same convention (which is reasonable), shall we do a check? (it will crash later on anyway)

Reply:
I think its worth doing the check here in the init. We have everything to make an informative error message so we might as well.